### PR TITLE
[SYCL][NFC] id.hpp & range.hpp operator tidyup

### PR DESCRIPTION
**Pull request by agozillon**
This should be a NFC, refactoring the hardcoded operators in the range 
and id classes into macros to generate them so that the classes are a 
little more concise to read.

### CODE REVIEW

**Comment by agozillon** [comment]
Not sure this is a great change, but seen that the id and range classes had a lot of handwritten operators and thought it might be possible to tidy them up into macros. 

That's the intent anyway, the changes don't break any of the existing tests on my machine and it compiles fine on my machine. So hopefully it passes the buildbot tests and is a somewhat useful change!

File: sycl/include/CL/sycl/id.hpp [hunk 1]
```diff
@@ -75,486 +75,77 @@ template <int dimensions = 1> struct id : public detail::array<dimensions> {
...
-    }
-    return *this;
-  }
-  id<dimensions> &operator^=(const id<dimensions> &rhs) {
-    for (int i = 0; i < dimensions; ++i) {
-      this->common_array[i] ^= rhs.common_array[i];
-    }
-    return *this;
-  }
+  #define GEN_OP(op)                                                           \
```
Comment (keryell) [inline sycl/include/CL/sycl/id.hpp]: Using such small macro names might cause some conflicts with some other libraries or user code.
What about using some reserved names, such as `__SYCL_GEN_OPT`? 
Note that this comment does apply to some other macros in some other parts of the current implementation that should be changed before up-streaming...

**Comment by bader** [comment]
@agozillon, could you squash your commits into single commit, please?

**Comment by agozillon** [comment]
No problem, it has been squashed.

**Pull request merged**